File: C++ Programming/problems on string manipulation/problem120.cpp
/*

120. Write a program which 2 strings from user and check whether first N contents of two strings are equal or
	    not. (Implement strcmp() function). Note : If third parameter is greater than the size of second string then concat whole string after first string. 
		
		Input :   “Marvellous Infosystems”    
					 “Marvellous Logic Building”    10
		
		Author : Milind Nalawade
	
		Date : 04/08/2021
	
		Language : C++ programming

*/
using namespace std;
#include<iostream>
#include<stdlib.h>
#include<string.h>

//Logic
int StrCmp(char* s1,char* s2,int no)
{
	int len1=strlen(s1);
	int len2=strlen(s2);
	int flag=1;
	
	for(int i=0;i<no;i++)
	{
		if(s1[i]!=s2[i])
		{
			flag=0;
			break;
		}
	}	
	return flag;
}

//Entry point function
int main(int argc,char* argv[])
{
	char* s1=argv[1];
	char* s2=argv[2];
	int no=atoi(argv[3]);
	int ret=0;
	ret=StrCmp(s1,s2,no);
	if(ret==1)
	{
		cout<<"\n TRUE \n";
	}
	else
	{
		cout<<"\n FALSE \n";
	}
	return 0;
}

File: Python Programming/problems on string manipulation/problem120.py
'''
120. Write a program which 2 strings from user and check whether first N contents of two strings are equal or
	    not. (Implement strcmp() function). Note : If third parameter is greater than the size of second string then concat whole string after first string. 
		
		Input :   “Marvellous Infosystems”    
					 “Marvellous Logic Building”    10
		
		Author : Milind Nalawade
	
		Date : 04/08/2021
	
		Language : Python 3.9.0
		
'''
import sys

#====
#Logic
#====
def StrCmp(s1,s2,no):
	flag=1
	for i in range(no):
		if i==len(s1) or i==len(s2):
			flag=0
			break	
		if s1[i]!=s2[i]:
			flag=0
	return flag
	
#=============
#Entry point function
#=============
def main():
	s1=sys.argv[1]
	s2=sys.argv[2]
	no=int(sys.argv[3])
	ret=int(StrCmp(s1,s2,no))
	if ret==1:
		print("\n TRUE \n")
	else:
		print("\n FLASE \n")
		
#=====
#starter
#=====
if __name__=="__main__":
	main()